new-log-viewer/src/services/LogFileManager.ts

```diff
@@ -0,0 +1,138 @@
+import {DecodeOptionsType} from "../typings/decoders";
+import {
+    CursorType,
+    FileSrcType,
+    LineNumLogEventNumMap,
+} from "../typings/worker";
+import {getUint8ArrayFrom} from "../utils/http";
+import {getBasenameFromUrl} from "../utils/url";
+import {
+    Decoder,
+    DecoderConstructor,
+} from "./decoders/Decoder";
+import JsonlDecoder from "./decoders/JsonlDecoder";
+
+
+/**
+ * A mapping of file name extensions to their respective decoders.
+ */
+const FILE_EXT_TO_DECODER: Readonly<Record<string, DecoderConstructor>> = Object.freeze({
+    ".jsonl": JsonlDecoder as unknown as DecoderConstructor,
+});
+
+class LogFileManager {
+    #pageSize: number;
+
+    #fileData: Uint8Array | null = null;
+
+    #fileName: string | null = null;
+
+    #numEvents: number = 0;
+
+    #decoder: Decoder | null = null;
+
+
+    /**
+     * Retrieves the decoder constructor based on the file extension.
+     *
+     * @param filename The filename from which to extract the file extension.
+     * @return The constructor associated with the file extension, or null if not found.
+     */
+    static #getDecoderConstructor = (filename: string): DecoderConstructor | null => {
+        for (const ext in FILE_EXT_TO_DECODER) {
+            if (Object.hasOwn(FILE_EXT_TO_DECODER, ext)) {
+                const decoderConstructor = FILE_EXT_TO_DECODER[ext];
+                if (filename.endsWith(ext) && "undefined" !== typeof decoderConstructor) {
+                    return decoderConstructor;
+                }
+            }
+        }
+
+        return null;
+    };
+
+    constructor (pageSize: number) {
+        this.#pageSize = pageSize;
+    }
+
+    async loadFile (fileSrc: FileSrcType): Promise<number> {
+        if ("string" === typeof fileSrc) {
+            this.#fileName = getBasenameFromUrl(fileSrc);
+            this.#fileData = await getUint8ArrayFrom(fileSrc, () => null);
+        } else {
+            // TODO: support file loading via Open / Drag-n-drop
+            console.error("Read from File not yet supported");
+        }
+
+        if (null === this.#fileData || null === this.#fileName) {
+            return 0;
+        }
+
+        const MatchingDecoder = LogFileManager.#getDecoderConstructor(this.#fileName);
+        if (null === MatchingDecoder) {
+            return 0;
+        }
+
+        this.#decoder = new MatchingDecoder(this.#fileData);
+        this.#numEvents = this.#decoder.buildIdx();
+        console.log(`Found ${this.#numEvents} log events.`);
+
+        return this.#numEvents;
+    }
+
+    setDecodeOptions (options: DecodeOptionsType) {
+        if (null === this.#decoder) {
+            throw new Error("loadFile() must be first called.");
+        }
+        this.#decoder.setDecodeOptions(options);
+    }
+
+    loadPage (cursor: CursorType): {
```

Comment:
```suggestion
    /**
    * Loads a page of log events based on the provided cursor.
    *
    * @param {CursorType} cursor - The cursor indicating the page to load. This can be a page number or other types of cursor.
    * @returns {Object} An object containing the logs as a string, a map of line numbers to log event numbers, and the cursor line number.
    * @throws {Error} - Throws an error if the loadFile method has not been called before this method.
    */
    loadPage (cursor: CursorType): {
```

Reply:
How about

/**
 * Loads a page of log events based on the provided cursor.
 *
 * @param cursor The cursor indicating the page to load. See {@link CursorType}.
 * @returns An object containing the logs as a string, a map of line numbers to log event numbers, and the line number of the first line in the cursor identified event.
 * @throws {Error} - Throws an error if the loadFile method has not been called before this method.
 */